Source/Core/Core/PowerPC/BreakPoints.h

```diff
@@ -132,3 +135,58 @@ class MemChecks
   Core::System& m_system;
   bool m_mem_breakpoints_set = false;
 };
+
+class DelayedMemCheckUpdate final
+{
+public:
+  DelayedMemCheckUpdate(MemChecks* memchecks, bool update_needed = false)
+      : m_memchecks(memchecks), m_update_needed(update_needed)
+  {
+  }
+
+  DelayedMemCheckUpdate(const DelayedMemCheckUpdate&) = delete;
+  DelayedMemCheckUpdate& operator=(const DelayedMemCheckUpdate&) = delete;
+
+  DelayedMemCheckUpdate(DelayedMemCheckUpdate&& other)
+      : m_memchecks(other.m_memchecks), m_update_needed(other.m_update_needed)
+  {
+    other.m_update_needed = false;
+  }
+
+  DelayedMemCheckUpdate& operator=(DelayedMemCheckUpdate&& other)
```

Comment:
Maybe the move-operator should be `delete`d, to force use of `operator|` ? It's maybe a little weird that moving would invoke `Update`?